sdk/src/system/Debugger/Client.ls

```diff
@@ -130,210 +130,226 @@ package system.debugger
                 socket.close();
 
             reloaded();
-        
+
+            return true;
         }
 
         /**
          *  Executes the program until the next breakpoint (if any).
          */
-        static function cmdRUN() {
-        
+        static function cmdRUN():Boolean {
+
             if (checkAssertionState())
-                return;
-            
+                return false;
+
             curFrameIdx = 0;
             Debug.blocking = false;
             Debug.stepping = false;
             Debug.stepOver = false;
             Debug.finishMethod = null;
-            
+
+            return true;
         }
 
         /**
          *  Sets a breakpoint at a given source file + line, example: setb src/demo/Program.ls 112.
          */
-        static function cmdBREAK(line:String) {
-        
-            if (checkAssertionState())
-                return;
-        
-            var commandv:Vector.<String> = line.find("^([A-Z]+)%s+(.-)%s+(%d+)%s*$");
+        static function cmdBREAK(line:String):Boolean {
 
-            Debug.addBreakpoint(commandv[1], commandv[2].toNumber());
+            if (checkAssertionState())
+                return false;
 
-            sendToServer("Setting breakpoint: " + commandv[1] + " : " + commandv[2].toNumber() + "\n");                
+            var commandv:Vector.<String> = line.find("^([A-Z]+)%s+(.-)%s+(%d+)%s*$");
 
-        
+            var result:String;
+            if((result = Debug.addBreakpoint(commandv[1], commandv[2].toNumber())) != null)
+            {
+                sendToServer("Setting breakpoint: " + result + " : " + commandv[2].toNumber() + "\n");
+                return true;
+            }
+            else
+            {
+                return false;
+            }
         }
-        
+
         /**
          *  Clear command.
          */
-        static function cmdCLEAR(line:String) {
-        
+        static function cmdCLEAR(line:String):Boolean {
+
             if (checkAssertionState())
-                return;
-        
+                return false;
+
             var commandv:Vector.<String> = line.find("^([A-Z]+)%s+(.-)%s+(%d+)%s*$");
```

Comment:
If this regex is the same as the one above, maybe we could have it up top as a constant string matcher?